lib/shared/layers/python-sdk/python/genai_core/admin_user_management.py

```diff
@@ -0,0 +1,185 @@
+import boto3
+import genai_core.auth
+import os
+
+COGNITO_USER_POOL_ID = os.environ.get("COGNITO_USER_POOL_ID")
+
+idp = boto3.client("cognito-idp")
+
+
+def create_user(name, email, role, phone_number=None):
+    if role in genai_core.auth.UserPermissions.VALID_ROLES:
+        attributes = [
+            {"Name": "email", "Value": email},
+            {"Name": "name", "Value": name},
+        ]
+        if phone_number:
+            attributes.append({"Name": "phone_number", "Value": phone_number})
+        idp.admin_create_user(
+            UserPoolId=COGNITO_USER_POOL_ID,
+            Username=email,
+            UserAttributes=attributes,
+        )
+        idp.admin_add_user_to_group(
+            UserPoolId=COGNITO_USER_POOL_ID,
+            Username=email,
+            GroupName=role,
+        )
+        return True
+    else:
+        raise Exception("Invalid role")
+
+
+def list_users():
+    response = idp.list_users(UserPoolId=COGNITO_USER_POOL_ID)
+    chatbot_users = idp.list_users_in_group(
+        UserPoolId=COGNITO_USER_POOL_ID, GroupName="chatbot_user"
+    )["Users"]
+    chatbot_workspaces_users = idp.list_users_in_group(
+        UserPoolId=COGNITO_USER_POOL_ID, GroupName="chatbot_workspaces_user"
+    )["Users"]
+    chatbot_workspaces_managers = idp.list_users_in_group(
+        UserPoolId=COGNITO_USER_POOL_ID, GroupName="chatbot_workspaces_manager"
+    )["Users"]
+    chatbot_admin = idp.list_users_in_group(
+        UserPoolId=COGNITO_USER_POOL_ID, GroupName="chatbot_admin"
+    )["Users"]
+    users = []
+    for user in response["Users"]:
```

Comment:
I think you can rewrite this loop to be more dynamic and less verbose considering the following:

- if you need to `snake_case` all attributes simply have a function `snake_case, loop over all attributes keys and call it - instead of hardcoded if statements.

- have a list of roles sorted by importance, loop over it, `if username in ...` and the first match will be the user role.

- let's use enums for roles from `permissions` instead of hardcoded strings